app/scenes/DocumentPermanentDelete.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+import { observer } from "mobx-react";
+import * as React from "react";
+import { useTranslation, Trans } from "react-i18next";
+import { useHistory } from "react-router-dom";
+import Document from "models/Document.js";
+import Button from "components/Button";
+import Flex from "components/Flex";
+import HelpText from "components/HelpText";
+import useStores from "hooks/useStores";
+import { collectionUrl } from "utils/routeHelpers";
+
+type Props = {
+  document: Document,
+  onSubmit: () => void,
+};
+
+function DocumentPermanentDelete({ document, onSubmit }: Props) {
+  const [isDeleting, setIsDeleting] = React.useState(false);
+  const { t } = useTranslation();
+  const { ui, documents, collections } = useStores();
+  const { showToast } = ui;
+  const history = useHistory();
+  const collection = collections.get(document.collectionId);
+
+  const handleSubmit = React.useCallback(
+    async (ev: SyntheticEvent<>) => {
+      ev.preventDefault();
+      try {
+        setIsDeleting(true);
+        await documents.delete(document, { permanent: true });
+        showToast(t("Document permanently deleted"), { type: "success" });
+        onSubmit();
+
+        if (documents.deleted.length) {
+          history.push("/trash");
+        } else {
+          history.push(collectionUrl(collection?.url || "/"));
```

Comment:
What's the thinking here? My immediate thought was "nice detail", but on further thought I think it would be better to stay in the trash and let the user see that it's empty?

Reply:
I thought of saving a click for the user 😝 . But, now that I think more, I agree it will be strange for the user to be navigated to a random collection.